locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/LocationNetworkInfoCustomizer.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.brooklyn.location.jclouds;
+
+import org.apache.brooklyn.util.core.config.ConfigBag;
+import org.jclouds.compute.domain.NodeMetadata;
+
+import com.google.common.annotations.Beta;
+
+/**
+ * LocationNetworkInfoCustomizers are used by {@link JcloudsLocation} to determine the host and port
+ * on which connections to locations should be made and the credentials that should be used.
+ *
+ * @see JcloudsLocationConfig#LOCATION_NETWORK_INFO_CUSTOMIZER
+ */
+@Beta
+public interface LocationNetworkInfoCustomizer extends JcloudsLocationCustomizer {
```

Comment:
Don't think this should be a `JcloudsLocationCustomizer`. Yes it customises the behaviour of the location. My thinking is that `JcloudsLocationCustomizer` is not properly named though - it's too generic and anything altering the behaviour of the location matches it. Something better describing it would be `LocationLifecycleCallback`.
To back the above - you are not implementing any of the methods of the customiser.

The primary task of this interface is to return the addresses we can ssh into. A name more fitting in this description would be `LocationConnectivityResolver`, `LocationAddressResolver`.